lib/src/builder_implementation.dart

```diff
@@ -4534,13 +4513,43 @@ String _extractMetadataCode(Element element, Resolver resolver,
       if (_isPrivateName(annotationNode.name.name)) {
         _severe("Cannot access private name ${annotationNode.name}", element);
       }
-      metadataParts.add("$prefix${annotationNode.name}");
+      String name = _extractNameWithoutPrefix(annotationNode.name, element);
+      if (_isPrivateName(name)) {
+        _severe("Cannot access private name $name", element);
+      }
+      metadataParts.add("$prefix$name");
     }
   }
 
   return _formatAsConstList("Object", metadataParts);
 }
 
+/// Extract the plain name from by [identifier] by stripping off the
+/// library import prefix at front, if any.
+String _extractNameWithoutPrefix(Identifier identifier, Element errorTarget) {
+  String name;
+  if (identifier is SimpleIdentifier) {
+    name = "$identifier";
```

Comment:
I worry about using `toString()` it is usually mostly a debugging tool. Is there not a `Identifier.token` or similar getter you could use returning a string? (Not that it is different from the existing code).